src/test/java/net/snowflake/client/jdbc/OauthAuthorizationCodeFlowLatestIT.java

```diff
@@ -0,0 +1,164 @@
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Tag(TestTags.CORE)
+public class OauthAuthorizationCodeFlowLatestIT extends BaseWiremockTest {
+
+  private static final String SCENARIOS_BASE_DIR = "/oauth/authorization_code";
+  private static final String SUCCESSFUL_FLOW_SCENARIO_MAPPINGS =
+      SCENARIOS_BASE_DIR + "/successful_scenario_mapping.json";
+  private static final String BROWSER_TIMEOUT_SCENARIO_MAPPING =
+      SCENARIOS_BASE_DIR + "/browser_timeout_scenario_mapping.json";
+  private static final String INVALID_SCOPE_SCENARIO_MAPPING =
+      SCENARIOS_BASE_DIR + "/invalid_scope_scenario_mapping.json";
+  private static final String TOKEN_REQUEST_ERROR_SCENARIO_MAPPING =
+      SCENARIOS_BASE_DIR + "/token_request_error_scenario_mapping.json";
+  private static final String CUSTOM_URLS_SCENARIO_MAPPINGS =
+      SCENARIOS_BASE_DIR + "/custom_urls_scenario_mapping.json";
+
+  private static final Logger logger =
+      LoggerFactory.getLogger(OauthAuthorizationCodeFlowLatestIT.class);
+
+  private final AuthExternalBrowserHandlers wiremockProxyRequestBrowserHandler =
+      new WiremockProxyRequestBrowserHandler();
+
+  @Test
+  public void successfulFlowScenario() throws SFException {
+    importMappingFromResources(SUCCESSFUL_FLOW_SCENARIO_MAPPINGS);
+    SFLoginInput loginInput =
+        createLoginInputStub("http://localhost:8009/snowflake/oauth-redirect", null, null);
+
+    OauthAccessTokenProvider provider =
+        new AuthorizationCodeFlowAccessTokenProvider(wiremockProxyRequestBrowserHandler, 30);
+    String accessToken = provider.getAccessToken(loginInput);
+
+    Assertions.assertFalse(StringUtils.isNullOrEmpty(accessToken));
+    Assertions.assertEquals("access-token-123", accessToken);
+  }
+
+  @Test
+  public void customUrlsScenario() throws SFException {
+    importMappingFromResources(CUSTOM_URLS_SCENARIO_MAPPINGS);
+    SFLoginInput loginInput =
+        createLoginInputStub(
+            "http://localhost:8007/snowflake/oauth-redirect",
+            String.format("http://%s:%d/authorization", WIREMOCK_HOST, wiremockHttpPort),
+            String.format("http://%s:%d/tokenrequest", WIREMOCK_HOST, wiremockHttpPort));
+
+    OauthAccessTokenProvider provider =
+        new AuthorizationCodeFlowAccessTokenProvider(wiremockProxyRequestBrowserHandler, 30);
+    String accessToken = provider.getAccessToken(loginInput);
+
+    Assertions.assertFalse(StringUtils.isNullOrEmpty(accessToken));
+    Assertions.assertEquals("access-token-123", accessToken);
+  }
+
+  @Test
+  public void browserTimeoutFlowScenario() {
```

Comment:
These error tests seem very similar, maybe we could make them parameterized? For keeping descriptive names, see `Arguments.argumentSet` (in case of `@ArgumentProvider`).

Reply:
I'd do it after client credentials flow is added, because I'm not yet sure if these will remain as they are.